api/versions.go

```diff
@@ -634,3 +657,16 @@ func handleVersionAPIErr(ctx context.Context, err error, w http.ResponseWriter,
 	log.Event(ctx, "request unsuccessful", log.ERROR, log.Error(err), data)
 	http.Error(w, err.Error(), status)
 }
+
+func checkVersion(ctx context.Context, version string) (int, error) {
+	versionId, err := strconv.Atoi(version)
+	if !(versionId > 0) {
+		log.Event(ctx, "version is not a positive integer", log.ERROR, log.Error(errs.ErrInvalidVersion), log.Data{"version": version})
+		return versionId, errs.ErrInvalidVersion
+	}
+	if err != nil {
```

Comment:
It makes sense to always check the error prior to anything else.

So if you switch the if statements around